lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -539,12 +546,20 @@ def scan(table_name, scan_hash, select_opts = {})
 
         Enumerator.new do |y|
           # Batch loop, pulls multiple requests until done using the start_key
+          # by paging in by batch size rather than limit size
+          eval_count = 0
           loop do
             results = client.scan(request)
-
             results.data[:items].each { |row| y << result_item_to_hash(row) }
 
-            if((lk = results[:last_evaluated_key]) && batch)
+            # If limit is set then keep track and stop requesting once
+            # completed. This is eval_limit so we compare to scanned_count
+            # rather than resulting data
+            eval_count += results.scanned_count
```

Comment:
That sounds good, so why don't we have:

* `eval_limit` - specifies the limit of returned records
* `scan_limit` - specifies the limit of scanned records (new)

In many cases these will act the same but in cases of filtering, `eval_limit` will continue until there are no more records or `eval_limit` is hit whereas the `scan_limit` will limit to basically one query (or multiple if used with `batch`).

Finally if `eval_limit` and `scan_limit` are used together, then we have following cases:

* `eval_limit = scan_limit` then our query `limit` will be the same
* `eval_limit > scan_limit` then our query `limit` will be the `scan_limit` (takes minimum)
* `eval_limit < scan_limit` then our query `limit` will be the `eval_limit` (takes minimum) but we stop once `scan_limit` is reached or when `eval_limit` is reached, whichever comes first.

If this sounds good, I can put this PR together to update to these semantics and add to README.

As for the `offset`, we already have that with `start` function, the only issue is things are okay if your result set includes your scanned set. However, if your scanned set is the super set then we need to return the `LastEvaluatedKey` somehow. It's not currently exposed by the `aws_sdk_v2` query/scan methods but dealing with this can be a separate issue.